pkg/sendconfig/sendconfig.go

```diff
@@ -0,0 +1,187 @@
+		return nil, fmt.Errorf("marshaling kong config into json: %w", err)
+	}
+
+	// fast path
+	if len(customEntitiesJSONBytes) == 0 {
+		return kongCoreConfig, nil
+	}
+
+	// slow path
+	mergeMap := map[string]interface{}{}
+	var result []byte
+	var customEntities map[string]interface{}
+
+	// unmarshal core config into the merge map
+	err = json.Unmarshal(kongCoreConfig, &mergeMap)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling kong config into map[string]interface{}: %w", err)
+	}
+
+	// unmarshal custom entities config into the merge map
+	err = json.Unmarshal(customEntitiesJSONBytes, &customEntities)
+	if err != nil {
+		// do not error out when custom entities are messed up
+		log.Errorf("failed to unmarshal custom entities from secret data: %v", err)
+	} else {
+		for k, v := range customEntities {
+			if _, exists := mergeMap[k]; !exists {
+				mergeMap[k] = v
+			}
+		}
+	}
+
+	// construct the final configuration
+	result, err = json.Marshal(mergeMap)
+	if err != nil {
+		err = fmt.Errorf("marshaling final config into JSON: %w", err)
+		return nil, err
+	}
+
+	return result, nil
+}
+
+func onUpdateInMemoryMode(ctx context.Context,
+	log logrus.FieldLogger,
+	state *file.Content,
+	customEntities []byte,
+	kongConfig *Kong,
+) error {
+	// Kong will error out if this is set
+	state.Info = nil
+	// Kong errors out if `null`s are present in `config` of plugins
+	deckgen.CleanUpNullsInPluginConfigs(state)
+
+	config, err := renderConfigWithCustomEntities(log, state, customEntities)
+	if err != nil {
+		return fmt.Errorf("constructing kong configuration: %w", err)
+	}
+
+	req, err := http.NewRequest("POST", kongConfig.URL+"/config",
+		bytes.NewReader(config))
+	if err != nil {
+		return fmt.Errorf("creating new HTTP request for /config: %w", err)
+	}
+	req.Header.Add("content-type", "application/json")
+
+	queryString := req.URL.Query()
+	queryString.Add("check_hash", "1")
+
+	req.URL.RawQuery = queryString.Encode()
+
+	_, err = kongConfig.Client.Do(ctx, req, nil)
+	if err != nil {
+		return fmt.Errorf("posting new config to /config: %w", err)
+	}
+
+	return err
+}
+
+func onUpdateDBMode(
+	targetContent *file.Content,
+	kongConfig *Kong,
+	selectorTags []string,
+) error {
+	// read the current state
+	rawState, err := dump.Get(kongConfig.Client, dump.Config{
+		SelectorTags: selectorTags,
+	})
+	if err != nil {
+		return fmt.Errorf("loading configuration from kong: %w", err)
+	}
+	currentState, err := state.Get(rawState)
+	if err != nil {
+		return err
+	}
+
+	// read the target state
+	rawState, err = file.Get(targetContent, file.RenderConfig{
+		CurrentState: currentState,
+		KongVersion:  kongConfig.Version,
+	})
+	if err != nil {
+		return err
+	}
+	targetState, err := state.Get(rawState)
+	if err != nil {
+		return err
+	}
+
+	syncer, err := diff.NewSyncer(currentState, targetState)
+	if err != nil {
+		return fmt.Errorf("creating a new syncer: %w", err)
+	}
+	syncer.SilenceWarnings = true
+	//client.SetDebugMode(true)
```

Comment:
should we remove this comment OR add more comments to it to provide context?

Reply:
Removed this comment - because the appropriate way to set `SetDebugMode` would be by injecting a client configured this way from `main()`.